### PR TITLE
Upgrade config

### LINKED ISSUE #2510: Upgrade server config
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. E.g. I wish I could use Milvus to do [...]

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Add any other context, code examples, or references to existing implementations about the feature request here.

### PR DESCRIPTION
**Pull request by wxyucs**
**What type of PR is this?**

Milvus config upgrade

**What this PR does / why we need it:**

**Which issue(s) this PR fixes:**

Fixes #2510 

**Special notes for your reviewer:**

**Additional documentation (e.g. design docs, usage docs, etc.):**

### CODE REVIEW

**Review by del-zhenwu — APPROVED** [review]

File: core/conf/demo/server_config.yaml [hunk 1]
```diff
@@ -9,115 +9,122 @@
...
+# auto_flush_interval  | The interval, in seconds, at which Milvus automatically    | Integer    | 1 (s)           |
+#                      | flushes data to disk.                                      |            |                 |
+#                      | 0 means disable the regular flush.                         |            |                 |
 #----------------------+------------------------------------------------------------+------------+-----------------+
-storage_config:
-  primary_path: /var/lib/milvus
-  secondary_path:
-  file_cleanup_timeout: 10
+storage:
+  path: /tmp/milvus
```
Comment (del-zhenwu) [inline core/conf/demo/server_config.yaml:68]: demo path need to update

**Review by jeffoverflow — APPROVED** [review]

**Review by JinHai-CN — APPROVED** [review]

**Review by yhmo — APPROVED** [review]

File: core/src/config/handler/GpuResourceConfigHandler.cpp [hunk 2]
```diff
@@ -94,19 +94,20 @@ GpuResourceConfigHandler::AddGpuSearchThresholdListener() {
     ConfigCallBackF lambda_gpu_threshold = [this](const std::string& value) -> Status {
         if (!gpu_enable_) {
             std::string msg =
-                std::string("GPU resources is disable. Cannot set config ") + CONFIG_ENGINE_GPU_SEARCH_THRESHOLD;
+                std::string("GPU resources is disable. Cannot set config ") + CONFIG_GPU_RESOURCE_GPU_SEARCH_THRESHOLD;
```
Comment (cydrain) [inline core/src/config/handler/GpuResourceConfigHandler.cpp:97]: disabled

**Review by tinkerlin — APPROVED** [review]

File: core/src/config/Config.cpp [hunk 3]
```diff
@@ -40,40 +41,61 @@ namespace server {
...
-const char* CONFIG_SERVER_PORT_DEFAULT = "19530";
-const char* CONFIG_SERVER_DEPLOY_MODE = "deploy_mode";
-const char* CONFIG_SERVER_DEPLOY_MODE_DEFAULT = "single";
-const char* CONFIG_SERVER_TIME_ZONE = "time_zone";
-const char* CONFIG_SERVER_TIME_ZONE_DEFAULT = "UTC+8";
-const char* CONFIG_SERVER_WEB_ENABLE = "web_enable";
-const char* CONFIG_SERVER_WEB_ENABLE_DEFAULT = "true";
-const char* CONFIG_SERVER_WEB_PORT = "web_port";
-const char* CONFIG_SERVER_WEB_PORT_DEFAULT = "19121";
+// const char* CONFIG_SERVER = "server_config";
```
Comment (cydrain) [inline core/src/config/Config.cpp:70]: can remove these all

File: core/src/config/Config.cpp [hunk 4]
```diff
@@ -119,21 +143,21 @@ const char* CONFIG_ENGINE_OMP_THREAD_NUM = "omp_thread_num";
...
 const char* CONFIG_GPU_RESOURCE_ENABLE = "enable";
 #ifdef MILVUS_GPU_VERSION
 const char* CONFIG_GPU_RESOURCE_ENABLE_DEFAULT = "true";
 #else
 const char* CONFIG_GPU_RESOURCE_ENABLE_DEFAULT = "false";
 #endif
-const char* CONFIG_GPU_RESOURCE_CACHE_CAPACITY = "cache_capacity";
-const char* CONFIG_GPU_RESOURCE_CACHE_CAPACITY_DEFAULT = "1";
+const char* CONFIG_GPU_RESOURCE_CACHE_CAPACITY = "cache_size";
+const char* CONFIG_GPU_RESOURCE_CACHE_CAPACITY_DEFAULT = "1073741824";
```
Comment (cydrain) [inline core/src/config/Config.cpp:156]: use 1024 * 1024 is better

File: core/src/config/Config.h [hunk 5]
```diff
@@ -28,40 +28,61 @@ using ConfigCallBackF = std::function<Status(const std::string&)>;
...
-extern const char* CONFIG_SERVER_PORT_DEFAULT;
-extern const char* CONFIG_SERVER_DEPLOY_MODE;
-extern const char* CONFIG_SERVER_DEPLOY_MODE_DEFAULT;
-extern const char* CONFIG_SERVER_TIME_ZONE;
-extern const char* CONFIG_SERVER_TIME_ZONE_DEFAULT;
-extern const char* CONFIG_SERVER_WEB_ENABLE;
-extern const char* CONFIG_SERVER_WEB_ENABLE_DEFAULT;
-extern const char* CONFIG_SERVER_WEB_PORT;
-extern const char* CONFIG_SERVER_WEB_PORT_DEFAULT;
+// extern const char* CONFIG_SERVER;
```
Comment (cydrain) [inline core/src/config/Config.h:57]: can remove these all

File: core/src/config/Config.cpp [hunk 6]
```diff
@@ -145,38 +169,30 @@ const char* CONFIG_TRACING = "tracing_config";
...
 const char* CONFIG_WAL_ENABLE = "enable";
 const char* CONFIG_WAL_ENABLE_DEFAULT = "true";
 const char* CONFIG_WAL_RECOVERY_ERROR_IGNORE = "recovery_error_ignore";
 const char* CONFIG_WAL_RECOVERY_ERROR_IGNORE_DEFAULT = "true";
 const char* CONFIG_WAL_BUFFER_SIZE = "buffer_size";
-const char* CONFIG_WAL_BUFFER_SIZE_DEFAULT = "256";
-const int64_t CONFIG_WAL_BUFFER_SIZE_MIN = 64;
-const int64_t CONFIG_WAL_BUFFER_SIZE_MAX = 4096;
-const char* CONFIG_WAL_WAL_PATH = "wal_path";
+const char* CONFIG_WAL_BUFFER_SIZE_DEFAULT = "268435456";
```
Comment (cydrain) [inline core/src/config/Config.cpp:178]: better to add some comments for these numbers

File: core/conf/server_config.template [hunk 7]
```diff
@@ -127,100 +134,55 @@ cache_config:
 #                      | if nq < gpu_search_threshold, the search computation will  |            |                 |
 #                      | be executed on both CPUs and GPUs.                         |            |                 |
 #----------------------+------------------------------------------------------------+------------+-----------------+
-engine_config:
-  use_blas_threshold: 1100
-  gpu_search_threshold: 1000
```
Comment (del-zhenwu) [inline core/conf/server_config.template:132]: gpu_search_threshold removed?

**Comment by del-zhenwu** [comment]
Set config value failed.
```
__________________________________________________________________________________________ TestStorageConfig.test_set_auto_flush_interval_valid __________________________________________________________________________________________

self = <test_config.TestStorageConfig object at 0x7f243ab605c0>, connect = <milvus.client.stub.Milvus object at 0x7f243ab9a518>, collection = 'test_0nlI1lY9'

    def test_set_auto_flush_interval_valid(self, connect, collection):
        '''
        target: set auto_flush_interval
        method: call set_config correctly
        expected: status ok, set successfully
        '''
        for valid_auto_flush_interval in [2, 1]:
            logging.getLogger().info(valid_auto_flush_interval)
            status, reply = connect.set_config("storage", "auto_flush_interval", valid_auto_flush_interval)
            assert status.OK()
            status, config_value = connect.get_config("storage", "auto_flush_interval")
            assert status.OK()
>           assert config_value == str(valid_auto_flush_interval)
E           AssertionError: assert '1' == '2'
E             - 1
E             + 2

test_config.py:1192: AssertionError
----------------------------------------------------------------------------------------------------------- Captured log call ------------------------------------------------------------------------------------------------------------
[2020-06-10 16:00:24-INFO-root]: 2 (test_config.py:1187)

```

**Comment by del-zhenwu** [comment]
```
self = <test_config.TestGeneralConfig object at 0x7f1939d9d208>, connect = <milvus.client.stub.Milvus object at 0x7f193a061a90>, collection = 'test_ljivM8e4'

    @pytest.mark.timeout(CONFIG_TIMEOUT)
    def test_get_time_zone_valid(self, connect, collection):
        '''
        target: get time_zone
        method: call get_config correctly
        expected: status ok
        '''
        status, config_value = connect.get_config("general", "time_zone")
>       assert status.OK()
E       AssertionError: assert False
E        +  where False = <bound method Status.OK of Status(code=1, message='Config node invalid: general.time_zone')>()
E        +    where <bound method Status.OK of Status(code=1, message='Config node invalid: general.time_zone')> = Status(code=1, message='Config node invalid: general.time_zone').OK

test_config.py:1037: AssertionError

````

**Comment by del-zhenwu** [comment]
```
self = <test_config.TestNetworkConfig object at 0x7fdaf7a39f98>, connect = <milvus.client.stub.Milvus object at 0x7fdaf7970668>, collection = 'test_PXaEdLak'

    @pytest.mark.timeout(CONFIG_TIMEOUT)
    def test_get_address_valid(self, connect, collection):
        '''
        target: get address
        method: call get_config correctly
        expected: status ok
        '''
        status, config_value = connect.get_config("network", "bind.address")
>       assert status.OK()
E       AssertionError: assert False
E        +  where False = <bound method Status.OK of Status(code=1, message='Invalid command: get_config network.bind.address')>()
E        +    where <bound method Status.OK of Status(code=1, message='Invalid command: get_config network.bind.address')> = Status(code=1, message='Invalid command: get_config network.bind.address').OK

test_config.py:829: AssertionError

```

**Comment by del-zhenwu** [comment]
```
(Pdb) connect._cmd("set_config gpu.search_threshold 1000")
(Status(code=1, message='Config node invalid: gpu.search_threshold'), None)
(Pdb) connect._cmd("set_config cache.insert_buffer_size 2GB")
(Status(code=0, message='Success!'), '')
(Pdb) connect._cmd("get_config cache.insert_buffer_size")
(Status(code=0, message='Success!'), '2GB')

```

**Review by JinHai-CN — APPROVED** [review]

**Pull request merged**